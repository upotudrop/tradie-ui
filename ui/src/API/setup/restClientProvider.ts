import axios, { AxiosInstance } from 'axios';

export const API_BASE_URL = 'http://localhost:8000';

export const restfulClient:AxiosInstance = axios.create({
    headers: {
        "Content-Type": 'application/json'
    }
})