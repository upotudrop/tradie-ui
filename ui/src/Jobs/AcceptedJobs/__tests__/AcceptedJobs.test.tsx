import axios from "axios";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { AcceptedJobs } from "..";
import { tempJobs } from "../../../__mocks__/tempJobs";
import MockAdapter from 'axios-mock-adapter';
import { API_BASE_URL } from "../../../API/setup/restClientProvider";

describe('Accepted Jobs testing suite', () => {
    beforeEach(() => {
        const mock = new MockAdapter(axios);
        mock.onGet(`${API_BASE_URL}/api/accepted-jobs`).reply(200, tempJobs)
    })
    
    it('should be able to render tradie ui', async () => {
        const { container } = await act(async () => render(<AcceptedJobs />))
        expect(container).toMatchSnapshot();
    })
    
    it('should render `no jobs available` if response empty', async () => {
        const mock = new MockAdapter(axios);
        mock.onGet(`${API_BASE_URL}/api/accepted-jobs`).reply(200, [])
    
        const { queryByText } = await act(async () => render(<AcceptedJobs />))
        expect(queryByText('No Jobs currently available')).toBeInTheDocument();
    }) 
})
