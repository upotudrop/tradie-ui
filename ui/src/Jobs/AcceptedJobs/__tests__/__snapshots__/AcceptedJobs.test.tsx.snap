// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accepted Jobs testing suite should be able to render tradie ui 1`] = `
<div>
  <section
    class="job-root-container"
  >
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          L
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Luke Skywalker
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Sydney
        </p>
        <p>
          2000
        </p>
        <p>
          Job ID: 
          1
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          luke@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Integer aliquam pulvinar odio et convallis. Integer id tristique ex. Aenean scelerisque massa vel est sollicitudin vulputate. Suspendisse quis ex eu ligula elementum suscipit nec a est. Aliquam a gravida diam. Donec placerat magna posuere massa maximus vehicula. Cras nisl ipsum, fermentum nec odio in, ultricies dapibus risus. Vivamus neque.
        </p>
      </section>
    </section>
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          D
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Darth Vader
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Bondi
        </p>
        <p>
          2026
        </p>
        <p>
          Job ID: 
          2
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          darth@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Praesent elit dui, blandit eget nisl sed, ornare pharetra urna. In cursus auctor tellus. Quisque ligula metus, viverra nec nibh ut, sagittis luctus tellus. Nulla egestas nibh ut diam vehicula, ut auctor lectus pharetra. Aliquam condimentum, erat eget vehicula eleifend, nulla risus consequat augue, quis convallis mi diam et dui.
        </p>
      </section>
    </section>
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          H
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Han Solo
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Manly
        </p>
        <p>
          2095
        </p>
        <p>
          Job ID: 
          3
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          han@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Aliquam posuere est sit amet libero egestas tempus. Donec ut efficitur sapien. Sed molestie nec lacus malesuada suscipit. Aliquam suscipit nibh at posuere tempor. Etiam a sollicitudin felis. In et enim leo. Morbi vel imperdiet purus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam posuere auctor elit, id venenatis.
        </p>
      </section>
    </section>
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          K
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Kylo Ren
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Surry Hills
        </p>
        <p>
          2010
        </p>
        <p>
          Job ID: 
          4
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          kylo@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Proin semper consectetur mauris id commodo. In accumsan est ligula, id posuere libero placerat ac. Nunc non volutpat sem. Mauris gravida dictum eleifend. Praesent quis mattis arcu, rutrum sagittis diam. Nullam tempus sagittis diam, vel viverra nunc ultricies non. Sed at orci sem. Phasellus eget arcu hendrerit, congue metus ut, mollis tellus. Quisque gravida metus ut libero porta, sit amet rutrum odio porta. Fusce interdum est sed quam venenatis dictum. Integer ultrices est in odio semper dictum. Proin nec urna vel quam finibus maximus.

Sed accumsan urna vitae libero luctus volutpat. Nulla eu sodales enim, vitae blandit ligula. Suspendisse at magna pellentesque, rhoncus orci quis, consequat diam. Donec pulvinar accumsan erat, quis hendrerit est ultricies vel. Vivamus felis justo, vulputate non urna sed, finibus semper ipsum. Cras mattis, est vel posuere mattis, turpis augue elementum massa, vitae accumsan nibh nisl nec lectus. Maecenas porta sagittis erat at consequat. Suspendisse fermentum rutrum bibendum. Donec tempor mollis massa vel egestas.

Morbi rutrum felis lacinia eros tincidunt scelerisque. Morbi aliquam porttitor sapien. Phasellus eu odio ac neque faucibus suscipit in at lectus. Maecenas et blandit arcu. Nullam sed sem neque. Nulla sit amet tristique nisl. Ut et pretium velit. Fusce consequat tincidunt fringilla. Nunc gravida libero sit amet augue viverra, a imperdiet odio dictum. Sed iaculis, metus vel rutrum convallis, quam ex commodo nibh, eget ultrices nisi eros eu massa. Ut iaculis maximus ligula, sed efficitur mauris bibendum sagittis. Curabitur et dolor mi. Proin lorem urna, porttitor quis lacus pharetra, ornare porta nulla. Sed ultricies feugiat nibh, et semper tellus aliquet sit amet. Cras faucibus scelerisque nisi, at vestibulum massa pharetra et.


        </p>
      </section>
    </section>
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          L
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Lando Calrissian
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Newtown
        </p>
        <p>
          2042
        </p>
        <p>
          Job ID: 
          5
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          lando@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Quisque blandit erat id mi tincidunt porta. Vivamus eleifend sagittis neque id maximus. Etiam molestie, massa ut tempus fermentum, augue nisi pulvinar nunc, id malesuada ipsum ipsum nec odio. Etiam et nisl facilisis, egestas massa eget, sagittis sapien. Curabitur eget consequat diam. Proin auctor rhoncus est, vitae imperdiet sem mollis.
        </p>
      </section>
    </section>
    <section
      class="job-card-container"
    >
      <section
        class="job-invitation-card-job-info-header"
      >
        <span
          class="job-owner-name-icon"
        >
          J
        </span>
        <section>
          <p
            class="job-owner-name"
          >
            Jabba TheHutt
             
          </p>
          November 11, 8:08 AM
        </section>
      </section>
      <section
        class="job-suburb-info-container"
      >
        <p>
          Sydney
        </p>
        <p>
          2000
        </p>
        <p>
          Job ID: 
          6
        </p>
      </section>
      <section
        class="flex-container"
      >
        <p
          class="highlighted-contact-info"
        >
          jabba@mailinator.com
        </p>
      </section>
      <section>
        <p>
          Suspendisse consequat malesuada arcu id venenatis. Donec maximus, dolor quis elementum scelerisque, lorem diam ornare arcu, sed venenatis orci justo nec nibh. Maecenas sollicitudin pulvinar lorem, at aliquet tortor tincidunt at. Vestibulum blandit, arcu eu blandit vehicula, orci nulla porta justo, a semper nunc risus sit amet ante. Donec lobortis.
        </p>
      </section>
    </section>
  </section>
</div>
`;
